### PR TITLE
As a user I should only be able to create a maximum of 3 targets

### PR DESCRIPTION
**Pull request by keinydev**
[Trello Card Here](https://trello.com/c/KCi0g1ya/5-2-as-a-user-i-should-only-be-able-to-create-a-maximum-of-3-targets)

### CODE REVIEW

File: spec/requests/api/v1/targets/create_spec.rb [hunk 1]
```diff
@@ -45,5 +45,29 @@
...
+        target_params[:target][:topic_id] = topic_list[1].id
+        post api_v1_targets_path, params: target_params, headers: auth_headers, as: :json
+      
+        target_params[:target][:title] = "Meet dog lovers"
+        target_params[:target][:topic_id] = topic_list[2].id
+        post api_v1_targets_path, params: target_params, headers: auth_headers, as: :json    
+
+        target_params[:target][:title] = "Meet pizza lovers"
+        target_params[:target][:topic_id] = topic_list[3].id
+        post api_v1_targets_path, params: target_params, headers: auth_headers, as: :json
```
Comment (santir489) [inline spec/requests/api/v1/targets/create_spec.rb:61]: This is the same as 
```
let(:topic_list) { create_list(3, :topics) } 
```
Right?

If you need topics of that list belong to the same user, you can do something like 
`let(:topic_list) { create_list(3, :topics, user: some_user) } `

File: app/models/target.rb [hunk 2]
```diff
@@ -3,4 +3,11 @@ class Target < ApplicationRecord
   belongs_to :user
 
   validates :title, :radius, :latitude, :longitude, :topic_id, presence: true
+  
+  validate :user_max_targets
+
+  def user_max_targets
+    user = User.find(self.user_id)
```
Comment (santir489) [inline app/models/target.rb:10]: You can remove this line. As this user is the user it self
Try this:
`errors.add(:targets, "You are allowed to create only 3 targets") if targets.length >= 3`

File: spec/models/target_spec.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  let(:topic_list) { create_list(:topic, 5) }   
+
+  let(:target_list) { create_list(:target, 3, user: user, topic: topic_list.first) }
+
+  let(:target) { Target.create(title: 'This cannot be created', radius: 5, latitude: 6.244203, longitude: -75.581215, topic_id: topic_list.first.id, user_id: user.id) }
+
+  describe 'callbacks' do 
+    context 'when user add more than 3 targets' do
+      before do
+        target_list
```
Comment (santir489) [inline spec/models/target_spec.rb:30]: If we need target_list to be created we can add `!` in `let` block like this:
`let!(:target_list) ...`

**Review by santir489 — APPROVED** [review]

File: app/models/target.rb [hunk 4]
```diff
@@ -1,6 +1,26 @@
...
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
 class Target < ApplicationRecord
   belongs_to :topic
   belongs_to :user
 
   validates :title, :radius, :latitude, :longitude, :topic_id, presence: true
+  
+  validate :user_max_targets
```
Comment (enzofab91) [inline app/models/target.rb:21]:
```suggestion
  validate :user_max_targets, on: :create
```

**Review by enzofab91 — APPROVED** [review]
Nice!

Left minor comment.

**Pull request merged**